### DISCUSSION TITLE
Create custom cluster for device

### DISCUSSION THREAD

**Discussion by samChanFarm66** [Koenkk/zigbee2mqtt#16786] [General]
Hi all, it is my first time to use zigbee2mqtt and I have some question - Is there any method to create custom cluster for device?

I have a zigbee module which cluster id: 0x08FC can be used.
I tried to edit the cluster.js file at "node_modules/zigbee-herdsman/dist/zcl/definition/cluster.js" , to add the cluster to it and it works perfectly.

However, I don't want to edit the source file. 
May I know is there any method to declare a custom cluster similar to add device by "external_converter.js" ? 
So cluster can add cluster externally.

Thank you

**Comment by Koenkk** [comment]
For what device is this? Currently clusters can only be added by adding it to `cluster.js`

**Reply by samChanFarm66** [reply]
I see, then I will use editing .js method to add custom cluster.
It is a Zigbee to Serial module, that cluster is to pass received payload to UART.

Thank you for your help.

**Comment by MarZab** [comment]
@Koenkk 

I have a related issue, I am working with a [device](https://smplproducts.com/) that uses the cluster 
 64515  0xFC03  but that is already defined as `manuSpecificPhilips2`

this device uses quite custom clusters and I also do not think they should be added to `cluster.js`. Is there a way to add the clusters per device or at least to disable the defualt handling ?

**Reply by Koenkk** [reply]
It can be added to cluster.js but with a manufacturer code (e.g. https://github.com/Koenkk/zigbee-herdsman/blob/master/src/zcl/definition/cluster.ts#L4165). Not that this only has to be done if you want to have commands/commands reponses, if you just want to support attributes it is not necessary to do this.
